src/mds/MDSMap.cc

```diff
@@ -1034,3 +1044,25 @@ bool MDSMap::is_degraded() const {
   }
   return false;
 }
+
+void MDSMap::set_min_compat_client(ceph_release_t version)
+{
+  vector<size_t> bits = CEPHFS_FEATURES_MDS_REQUIRED;
+
+  if (version >= ceph_release_t::octopus)
+    bits.push_back(CEPHFS_FEATURE_OCTOPUS);
+  else if (version >= ceph_release_t::nautilus)
+    bits.push_back(CEPHFS_FEATURE_NAUTILUS);
+  else if (version >= ceph_release_t::mimic)
+    bits.push_back(CEPHFS_FEATURE_MIMIC);
+  else if (version >= ceph_release_t::luminous)
+    bits.push_back(CEPHFS_FEATURE_LUMINOUS);
+  else if (version >= ceph_release_t::kraken)
+    bits.push_back(CEPHFS_FEATURE_KRAKEN);
+  else if (version >= ceph_release_t::jewel)
+    bits.push_back(CEPHFS_FEATURE_JEWEL);
```

Comment:
Shouldn't a minimum version of OCTOPUS also require the NAUTILUS feature bit, etc?

Reply:
not sure. I just move code from Server::update_required_client_features() to here. not change behavior